pulser/sequence.py

```diff
@@ -638,39 +657,30 @@ def add(
                     if op.targets & last.targets or protocol == "wait-for-all":
                         current_max_t = op.tf
                         break
-        ti = current_max_t
-        if ti > t0:
-            # Insert a delay
-            delay_duration = ti - t0
 
+        delay_duration = current_max_t - t0
+        # Find last pulse and compare phase
+        for op in self._schedule[channel][::-1]:
+            if isinstance(op.type, Pulse):
+                if op.type.phase != pulse.phase:
+                    delay_duration = max(
+                        delay_duration,
+                        # Considers that the last pulse might not be at t0
+                        self._channels[channel].phase_jump_time - (t0 - op.tf),
+                    )
+                break
+
+        if delay_duration > 0:
             # Delay must not be shorter than the min duration for this channel
             min_duration = self._channels[channel].min_duration
             if delay_duration < min_duration:
-                ti += min_duration - delay_duration
                 delay_duration = min_duration
```

Comment:
Could be simplified to `delay_duration = max(delay_duration, min_duration)`

Reply:
Indeed it can.